gwpy/timeseries/timeseries.py

```diff
@@ -1920,32 +1931,30 @@ def q_transform(self, qrange=(4, 64), frange=(0, numpy.inf),
 
         # build regular Spectrogram from peak-Q data by interpolating each
         # (Q, frequency) `TimeSeries` to have the same time resolution
-        nx = int(abs(Segment(*outseg)) / tres)
+        nx = int(abs(outseg) / tres)
         ny = frequencies.size
         out = Spectrogram(numpy.zeros((nx, ny)), x0=outseg[0], dx=tres,
                           frequencies=frequencies)
-        # FIXME: bug in Array2D.yindex setting
-        out._yindex = type(out.y0)(frequencies, out.y0.unit)
         # record Q in output
         out.q = peakq
         # interpolate rows
+        xout = numpy.linspace(0, 1, num=nx)
         for i, row in enumerate(norms):
-            interp = InterpolatedUnivariateSpline(
-                row.times.value, row.value)
-            out[:, i] = interp(out.times.value)
-
-        # then interpolate the spectrogram to increase the frequency resolution
-        # --- this is done because duncan doesn't like interpolated images
-        #     because they don't support log scaling
```

Comment:
@alurban, were these comments intentionally removed?

Reply:
@duncanmmacleod, yeah, these were intentionally removed. I was only trying to peacefully remove clutter, but if you want them put back I am happy to do that.

Reply:
yes, please replace them.

Reply:
@duncanmmacleod, this is done in the latest commit.